submissions/dashakim/oop-task/js/app.js

```diff
@@ -0,0 +1,106 @@
+var gameScore = 0;
+
+const ENEMY = {
+  speed: 100,
+  varianceSpeed: 222,
+  location: [70, 155, 235]
+};
+const PLAYER = {
+  startX: 200,
+  startY: 404,
+  stepX: 101,
+  stepY: 83
+};
+
+const CANVAS = {
+  begin: -60,
+  end: 510,
+  leftSide: 0,
+  rightSide: 400,
+  top: 0,
+  bottom: 404,
+  halfCell: 50
+};
+
+var Draw = function() {};
+
+Draw.prototype.render = function() {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
```

Comment:
`Draw.render` refers to the properties `Draw` class doesn't have. It is counter Object Oriented Programming principles for parent class to refer the props that only children classes have. Children can refer props they inherit from parent class. Why that? When you have a deep hierarchy of classes maintenance would become hell. [Encapsulation](https://medium.com/@cancerian0684/what-are-four-basic-principles-of-object-oriented-programming-645af8b43727) and [Separation of Concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) principles are to rescue here.

Luckily these props are common for both `Enemy` and `Player` so we can refactor appropriately.
I would also call the class `Character` or something like this so it looks like a real taxonomy.

Any child class can initialize its parent. Here is the pattern to follow:
```
var Figure = function (x, y, figureName) { this.x = x; this.y = y; this.name = figureName; }
Figure.prototype.constructor = Figure;
Figure.prototype.drawCenter = function () { /* ..... */ }

var Circle = function (x, y, radius) {
  Figure.call(this, x, y, 'Circle'); // some arguments are processed by parent constructor
  this.radius = radius;
}
Circle.prototype = Object.create(Figure.prototype);
Circle.prototype.constructor = Circle;
Circle.prototype.draw = function () {
  this.drawCenter(); // this method is inherited
  /* code to draw circle per se */
}
```

Reply:
I think I did it the same as previous version. Except now I'm understand that method "call" it's like "super" for ES6. Please check if I did it like you suggested. By the way, thanks for  the detailed explanation!